### DISCUSSION TITLE
Positional records bug? Parent class has positional member, child doesn't, gives syntax error

### DISCUSSION THREAD

**Discussion by canton7** [dotnet/csharplang#3795] [Q&A] [answered]
It seems there's something weird going on when a base positional record has a member, and a derived record attempts to pass a value and doesn't have any positional members of its own.

```cs
public record Base(string S) { }
public record Derived : Base("S") { }
//            ~~~~~~~       ~
//               |          |
//               |          error CS8861: Unexpected argument list.
//               error CS1729: 'Base' does not contain a constructor that takes 0 arguments
```

Giving the derived type any positional members fixes the issue:

```cs
public record Base(string S) { }
public record Derived(string T) : Base("S") { }
// No errors
```

[SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGALAbgFgAoAYgDsBXAG2omGrgznPsZJJQGYME4BjAPYIAJhgBC0OAAocmAMoBKDAG8MAX048+Q0RgByggOrCA1gE9Z2TABVlICVOkAiec+VrNxbrwHCxJggWGA6SUDKu7qoaHMSsYHBQAA4Q/Ew4AHQASpTkMACWCRkAwoJgSfmMCPKIAG75aVAkKiQYbVg8OABsWABMGACSUACiAB7wCKzUA+T5MNFeXkA===)

**Comment by RikkiGibson** [comment, marked as answer]
From the [specification](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/records.md):

> It is an error for a record to provide a record_base argument_list if the record_declaration does not contain a parameter_list.

I do not recall why this limitation was introduced, but the behavior you're seeing is the design. You would need to instead do something like

```cs
public record Base(string S) { }
public record Derived : Base
{
    public  Derived() : base("S") { }
}
```

**Comment by andresmoschini** [comment]
@agocke I saw that you have introduced this change: https://github.com/dotnet/csharplang/pull/3527/files#diff-584f61a480e3d35cb05945dcd5be8e3eR28

For sure there is a good reason, that I cannot understand, could you explain us the rationale behind this restriction?

**Comment by PathogenDavid** [comment]
@andresmoschini LDM [already made the decision to allow an empty parameter list](https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-08-24.md#base-calls-on-parameterless-records), but the spec hasn't been updated yet.

[It's already allowed in master](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAN0QSwGYE8kAmIA1AD4ACAzAAQJwDGA9ggdQELRwAU5AjAAzUAygEpqAb2oBfALAAoKrQbNWAESwYCXMSHacuAIiEGxk2XPkA7CGDhQADhHpxqfAHQAlAK6WYmW24Awoxg9pgANohCiGiYzlASrgBM1ACSUACiAB7wCNbhqZaYMIlS0vJAA==). (Done in https://github.com/dotnet/csharplang/pull/3900)

**Comment by andresmoschini** [comment]
Thanks @PathogenDavid! your answer is very useful for me to learn were to read.
